ocs_ci/ocs/node.py

```diff
@@ -982,9 +983,19 @@ def delete_and_create_osd_node_vsphere_upi_lso(osd_node_name, use_existing_node=
     old_pv_objs = get_pv_objs_in_sc(sc_name)
 
     osd_node = get_node_objs(node_names=[osd_node_name])[0]
-    osd_pod = get_node_pods(osd_node_name, pods_to_search=pod.get_osd_pods())[0]
-    osd_id = pod.get_osd_pod_id(osd_pod)
-    log.info(f"osd id to remove = {osd_id}")
+    osd_ids = get_node_osd_ids(osd_node_name)
+    assert osd_ids, f"The node {osd_node_name} does not have osd pods"
+
+    ocs_version = config.ENV_DATA["ocs_version"]
+    if len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6"):
+        log.warning(
+            f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
+            f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
+        )
+        assert False
+
```

Comment:
```suggestion
    ocs_version = config.ENV_DATA["ocs_version"]
    assert len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6"), (
        f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
        f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
    )

```

Reply:
I think you meant `assert not (
        len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6")
    ), (
        f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
        f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
    )`. We need to exclude this condition

Reply:
thanks for checking. It's ok to keep it as you did initially

Reply:
ok, np